test/idle-source-e2e/testdata/idle-source-reduce-pipeline.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: idle-source
+spec:
+  watermark:
+    idleSource:
+      threshold: 5s
+      incrementBy: 3s
+      stepInterval: 2s
```

Comment:
Can we add some comments on these numbers on how they are used to test source idling? The comment should make it clear why removing these configurations will break the test.

Reply:
I am un-resolving this one. We configured the idle source, updated the simple keyed reduce test with a new start time and new wait time before sending next batch.

I have a hard time understanding how this test works in terms of verifying idle source works. Can we write some comments on why this test is testing the source idling functionality? How these numbers `5s`, `3s`, `2s`, `startTime := 100, `time.Sleep(10 * time.Millisecond)` and `WithTimeout(120*time.Second)` work together to achieve idle source test?

Reply:
#1409